comp/logs/agent/agent.go

```diff
@@ -278,3 +290,57 @@ func (a *agent) GetMessageReceiver() *diagnostic.BufferedMessageReceiver {
 func (a *agent) GetPipelineProvider() pipeline.Provider {
 	return a.pipelineProvider
 }
+
+func (a *agent) onUpdateSDSRules(updates map[string]state.RawConfig, applyStateCallback func(string, state.ApplyStatus)) { //nolint:revive
+	var err error
+
+	for _, config := range updates {
+		if rerr := a.pipelineProvider.ReconfigureSDS(config.Config, nil); rerr != nil {
+			err = rerr
+		}
+	}
+
+	if err != nil {
+		log.Errorf("Can't update SDS standard rules: %v", err)
+	}
+
+	// Apply the new status to all configs
+	for cfgPath := range updates {
+		if err == nil {
+			applyStateCallback(cfgPath, state.ApplyStatus{State: state.ApplyStateAcknowledged})
+		} else {
+			applyStateCallback(cfgPath, state.ApplyStatus{
+				State: state.ApplyStateError,
+				Error: err.Error(),
```

Comment:
Just making sure that the error string here is something you'd be ok with being visible to users.